plugin/trino-hive/src/main/java/io/trino/plugin/hive/RecordFileWriter.java

```diff
@@ -98,11 +100,13 @@ public RecordFileWriter(
 
         String serDe = storageFormat.getSerDe();
         serializer = initializeSerializer(conf, schema, serDe);
-        recordWriter = createRecordWriter(path, conf, schema, storageFormat.getOutputFormat(), session);
 
         List<ObjectInspector> objectInspectors = getRowColumnInspectors(fileColumnTypes);
         tableInspector = getStandardStructObjectInspector(fileColumnNames, objectInspectors);
 
+        TextHeaderWriter textHeaderWriter = new TextHeaderWriter(serializer, typeManager, session, fileColumnNames);
```

Comment:
Please create `THW` conditionally only if `outputFormatName.equals(HiveIgnoreKeyTextOutputFormat.class.getName())` and add `checkArguement in `HWU.createRecordWriter` that `Optional<TextHeaderWriter> textHeaderWriter` is non-empty only in matching switch branch.

Actually I would prefer to create `TextHeaderWriter` in `HWU.createRecordWriter` but it feels like you would need to pass lots of stuff there.

Reply:
ok. I had similar idea and concerns regarding creating `TextHeaderWriter` in `HWU`